File: java/acronym/src/main/java/Acronym.java
class Acronym {
    String phrase;

    Acronym(String phrase) {
        this.phrase = phrase;
    }

    String get() {
        phrase = phrase.replace("-"," ").replace("_", " ");
        String [] tokens= phrase.trim().split(" ");
        StringBuilder acronym = new StringBuilder();

        for (String token : tokens){
            if(!token.trim().isEmpty()){
                acronym.append(token.toUpperCase().charAt(0));
            }
        }
        return acronym.toString();
    }

}

File: kotlin/acronym/src/main/kotlin/Acronym.kt
object Acronym {
    fun generate(phrase: String) : String {
        val newPhrase = phrase.replace("-", " ").replace("_", " ")
        var tokens = newPhrase.split(" ")
        println(tokens)
        var acronym = StringBuilder()

        for (token in tokens){
            if (token.trim().isNotEmpty()){
                acronym.append(token.uppercase()[0])
            }
        }
        return acronym.toString()
    }
}
